iceoryx_posh/include/iceoryx_posh/internal/popo/client_impl.inl

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_POSH_POPO_CLIENT_IMPL_INL
+#define IOX_POSH_POPO_CLIENT_IMPL_INL
+
+#include "iceoryx_posh/internal/popo/client_impl.hpp"
+
+namespace iox
+{
+namespace popo
+{
+template <typename Req, typename Res, typename BaseClientT>
+ClientImpl<Req, Res, BaseClientT>::ClientImpl(const capro::ServiceDescription& service,
+                                              const ClientOptions& clientOptions) noexcept
+    : BaseClientT(service, clientOptions)
+{
+}
+
+template <typename Req, typename Res, typename BaseClientT>
+cxx::expected<Request<Req>, AllocationError> ClientImpl<Req, Res, BaseClientT>::loanUninitialized() noexcept
+{
+    auto result = port().allocateRequest(sizeof(Req), alignof(Req));
+    if (result.has_error())
+    {
+        return cxx::error<AllocationError>(result.get_error());
+    }
+    else
```

Comment:
It is not a hard must but some compiler may throw a warning here since you have no return at the end of the function even when this covers all cases.
Would you mind to do just an early return here and remove the `else` and just put the code after the early return